components/gitpod-cli/pkg/supervisor/ports.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package supervisor
+
+import (
+	"context"
+	"io"
+
+	"github.com/gitpod-io/gitpod/supervisor/api"
+	log "github.com/sirupsen/logrus"
+)
+
+func GetPortsList(ctx context.Context, client api.StatusServiceClient) []*api.PortsStatus {
+	listen, err := client.PortsStatus(ctx, &api.PortsStatusRequest{Observe: false})
+	if err != nil {
+		log.WithError(err).Error("Cannot list ports")
+	}
+
+	portschan := make(chan []*api.PortsStatus)
+
+	go func() {
+		for {
+			resp, err := listen.Recv()
+			if err != nil && err != io.EOF {
+				panic(err)
+			}
+
+			chunk := resp.GetPorts()
+
+			if chunk == nil {
+				close(portschan)
+				break
+			} else {
+				portschan <- chunk
+			}
+		}
+	}()
+
+	return <-portschan
```

Comment:
```suggestion
	srv, err := client.PortsStatus(ctx, &api.PortsStatusRequest{Observe: false})
	if err != nil {
		log.WithError(err).Error("Cannot get ports status service")
	}
	resp, err := srv.Recv()
	if err != nil {
		log.WithError(err).Error("Cannot get ports status")
	}
	return resp.GetPorts()
```

Reply:
We should process errors...I missed it. 😅

Option 1:
1. Wrap errors in `GetPortsList` func and return as second result
2. Get, check, and output  error outside

Option 2:
1. Move func `GetPortsList` content inside command, log errors and return here